pkg/block/block_linux.go

```diff
@@ -329,29 +331,36 @@ func disks(ctx *context.Context, paths *linuxpath.Paths) []*Disk {
 		wwn := diskWWN(paths, dname)
 		removable := diskIsRemovable(paths, dname)
 
-		d := &Disk{
-			Name:                   dname,
-			SizeBytes:              size,
-			PhysicalBlockSizeBytes: pbs,
-			DriveType:              driveType,
-			IsRemovable:            removable,
-			StorageController:      storageController,
-			BusPath:                busPath,
-			NUMANodeID:             node,
-			Vendor:                 vendor,
-			Model:                  model,
-			SerialNumber:           serialNo,
-			WWN:                    wwn,
-		}
+		if driveType == DRIVE_TYPE_LOOP && size == 0 {
+			// Do nothing for empty loop devices
+			// There can be several loop devices present on /dev but that doesnt mean they are in use
+			// With this we skip any non-used loop devices
+		} else {
```

Comment:
I would prefer to just return all the loop devices that are listed by the OS, regardless of their size. Calling programs can filter out returned results any way they want.

Reply:
Thing is, if you mount something on a loop device, several loop devices will appear in /dev, even if not mounted. This was a kind fo fix to report only those loop devices in use.

```
$ sudo losetup -f --show cOS-Seed-green-0.8.8-g8522da85-x86_64.iso 
/dev/loop0

$ ls /dev/loop*
/dev/loop0  /dev/loop2  /dev/loop4  /dev/loop6  /dev/loop-control
/dev/loop1  /dev/loop3  /dev/loop5  /dev/loop7

$ sudo blkid |grep loop
/dev/loop0: BLOCK_SIZE="2048" UUID="2022-03-23-20-47-00-00" LABEL="COS_LIVE" TYPE="iso9660" PTTYPE="dos"
```

So in reality, in this situation, only the loop0 device is being used, the rest are there for some reason I guess, but not used.  Returning unused loop devices seemed to me a waste of user time filtering those out as they are never gonna need them for anything.

Thoughts?

Reply:
@Itxaka OK, I guess I can get behind that reasoning :)

Do me a favor though... in order to reduce the number of lines of code changed here, can you edit the above to this instead?

```go
if storageController == STORAGE_CONTROLLER_LOOP && size == 0 {
    // We don't care about unused loop devices...
    continue
}
```

and please place above the `d := &Disk{...}` definition line.